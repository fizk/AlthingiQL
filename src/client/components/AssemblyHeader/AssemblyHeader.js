import React from 'react';
import PropTypes from 'prop-types';
import {H1} from "../../elements/Headline/index";
import {Badge} from "../../elements/Badge/index";
import {Grid, Row, Column} from "../../elements/Grid/index";

export default class AssemblyHeader extends React.Component {
    static propTypes = {
        assembly: PropTypes.shape({
            id: PropTypes.number,
            period: PropTypes.shape({
                from: PropTypes.string,
                to: PropTypes.string,
            }),
            division: PropTypes.shape({
                majority: PropTypes.arrayOf(PropTypes.shape({
                    id: PropTypes.number,
                    name: PropTypes.string,
                    color: PropTypes.string,
                })),
                minority: PropTypes.arrayOf(PropTypes.shape({
                    id: PropTypes.number,
                    name: PropTypes.string,
                    color: PropTypes.string,
                })),
            }),
        }),
        loading: PropTypes.bool,
    };

    static defaultProps = {
        assembly: {
            id: undefined,
            period: {
                from: undefined,
                to: undefined,
            },
            division: {
                majority: [],
                minority: []
            }
        },
        loading: false,
    };

    render() {
        return (
            <Grid>
                <Row>
                    <Column>
                        <H1>{this.props.assembly.id}</H1>
                        <time>
                            {this.props.assembly.period.from}
                            {this.props.assembly.period.to}
                        </time>
                    </Column>
                    <Column>
                        <Row>
                            {this.props.assembly.division.majority.map(party => (
                                <Column key={`party-${party.id}`}>
                                    <Badge title={party.name} color={party.color}/>
                                </Column>
                            ))}
                        </Row>
                        <Row>
                            {this.props.assembly.division.minority.map(party => (
                                <Column key={`party-${party.id}`}>
                                    <Badge title={party.name} color={party.color}/>
                                </Column>
                            ))}
                        </Row>
                    </Column>
                </Row>
            </Grid>
        )
    }
}

