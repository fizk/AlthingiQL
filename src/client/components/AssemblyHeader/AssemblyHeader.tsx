import * as React from "react";
import { H1 } from "../../elements/Headline";
import Badge from "../../elements/Badge";
import './index.scss';
import {Link} from "react-router-dom";

type AssemblyHeaderProps = {
    assembly?: {
        id?: number,
        period?: {
            from?: string,
            to?: string
        },
        division?: {
            majority?: {
                id?: number,
                name?: string,
                color?: string
            }[],
            minority?: {
                id?: number,
                name?: string,
                color?: string
            }[]
        }
    },
    loading?: boolean
};

export default class AssemblyHeader extends React.Component<AssemblyHeaderProps, {}> {
    static defaultProps = {
        assembly: {
            id: undefined,
            period: {
                from: undefined,
                to: undefined
            },
            division: {
                majority: [],
                minority: []
            }
        },
        loading: false
    };

    render() {
        return (
            <div className="assembly-header">
                <div className="assembly-header__headline">
                    <H1>
                        <Link to={`/loggjafarthing/${this.props.assembly.id}`}>
                            {this.props.assembly.id}
                        </Link>. Löggjafarþing
                    </H1>
                    <time>
                        {this.props.assembly.period.from}
                        {this.props.assembly.period.to}
                    </time>
                </div>
                <div className="assembly-header__parties">
                    <ul className="assembly-header__party-list">
                        {this.props.assembly.division.majority.map(party => (
                            <li key={`party-${party.id}`}
                                className="assembly-header__party-list-item">
                                <Badge title={party.name} color={party.color} />
                            </li>
                        ))}
                    </ul>
                    <ul className="assembly-header__party-list">
                        {this.props.assembly.division.minority.map(party => (
                            <li key={`party-${party.id}`}
                                className="assembly-header__party-list-item">
                                <Badge title={party.name} color={party.color} />
                            </li>
                        ))}
                    </ul>
                </div>
            </div>
        );
    }
}
