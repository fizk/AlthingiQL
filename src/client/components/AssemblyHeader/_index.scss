@import "../../config";

.assembly-header {
    display: flex;
    padding: map_get($space, xl) 0;
}

.assembly-header__headline {
    flex-grow: 1;
}

.assembly-header__parties {

}

.assembly-header__party-list {
    display: flex;
    justify-content: flex-end;
}

.assembly-header__party-list-item {
    padding: map_get($space, xxs) / 2;
}
