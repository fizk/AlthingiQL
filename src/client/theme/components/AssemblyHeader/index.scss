.assembly-header {
    color: currentColor;
    padding: var(--space-l);
}

.assembly-header__title {
    font-size: var(--font-size-l);
}

.assembly-header__sub-title {
    font-size: var(--font-size-m);
    font-weight: 300;
}
