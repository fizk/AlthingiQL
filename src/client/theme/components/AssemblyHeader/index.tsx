import * as React from 'react';
import {FunctionComponent} from 'react';
import Menu, {MenuItem} from '../../elements/Menu';
import './index.scss';

interface Props {
    id: number;
    title: string;
}

const AssemblyHeader: FunctionComponent<Props> = ({id, title}) => (
    <div className="assembly-header">
        <h2 className="assembly-header__title">{id}. Löggjafarþing</h2>
        <h3 className="assembly-header__sub-title">{title}</h3>
        <Menu>
            <MenuItem exact to={`/loggjafarthing/${id}`}>Samantekt</MenuItem>
            <MenuItem to={`/loggjafarthing/${id}/thingmal`}>Þingmál</MenuItem>
            <MenuItem to={`/loggjafarthing/${id}/thingfundir`}>Þingfundir</MenuItem>
            <MenuItem to={`/loggjafarthing/${id}/thingmenn`}>Þingmenn</MenuItem>
            <MenuItem to={`/loggjafarthing/${id}/thingflokkar`}>Þingflokkar</MenuItem>
        </Menu>
    </div>
);

export default AssemblyHeader;
