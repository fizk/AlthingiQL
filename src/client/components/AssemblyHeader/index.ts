import {graphql} from 'react-apollo';
import compose from '../../utils/compose';
import gql from 'graphql-tag';
import AssemblyHeader from './AssemblyHeader';

const assemblyQuery = gql`
    query assembly ($assembly: Int!) {
        Assembly(assembly: $assembly) {
            id
            period {
                from
                to
            }
            division {
                majority {
                    id
                    name
                    color
                }
                minority {
                    id
                    name
                    color
                }
            }
            cabinet {
                id
                title
                period {from to}
            }
        }
    }
`;

export default compose(
    graphql(assemblyQuery, {
        props: ({data: {loading, Assembly}}: any) => ({
            assembly: loading === false ? Assembly : undefined,
            loading: loading,
        }),
        options: ({assembly}: any) => ({
            variables: {
                assembly,
            },
        }),
    }),
)(AssemblyHeader);
