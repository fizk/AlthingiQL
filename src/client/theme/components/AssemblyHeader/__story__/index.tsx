import * as React from 'react';
import {storiesOf} from '@storybook/react';
import {BrowserRouter as Router} from 'react-router-dom';
import {withKnobs} from '@storybook/addon-knobs';
import AssemblyHeader from '../index'
import '../../../global.scss';

const stories = storiesOf('Theme/AssemblyHeader', module);
stories.addDecorator(withKnobs);

stories.add('index', () => {
    return (
        <Router>
            <AssemblyHeader id={149} title="RÃ¡Ã°uneyti KatrÃ­nar JakobsdÃ³ttur" />
        </Router>
    )
});
