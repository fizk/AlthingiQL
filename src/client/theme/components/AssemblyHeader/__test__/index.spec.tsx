import * as React from 'react';
import AssemblyHeader from '../index';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

describe('<AssemblyHeader />', () => {
    test('default props', () => {
        const tree = renderer.create(
            <Router >
                <AssemblyHeader id={1} title={''} />
            </Router>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
