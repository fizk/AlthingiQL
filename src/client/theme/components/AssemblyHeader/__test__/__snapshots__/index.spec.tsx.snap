// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssemblyHeader /> default props 1`] = `
<div
  className="assembly-header"
>
  <h2
    className="assembly-header__title"
  >
    1
    . Löggjafarþing
  </h2>
  <h3
    className="assembly-header__sub-title"
  >
    
  </h3>
  <nav
    className="menu__nav"
  >
    <ul
      className="menu__menu"
    >
      <li
        className="menu__menu-item"
      >
        <a
          aria-current={null}
          className="menu__menu-link"
          href="/loggjafarthing/1"
          onClick={[Function]}
        >
          Samantekt
        </a>
      </li>
      <li
        className="menu__menu-item"
      >
        <a
          aria-current={null}
          className="menu__menu-link"
          href="/loggjafarthing/1/thingmal"
          onClick={[Function]}
        >
          Þingmál
        </a>
      </li>
      <li
        className="menu__menu-item"
      >
        <a
          aria-current={null}
          className="menu__menu-link"
          href="/loggjafarthing/1/thingfundir"
          onClick={[Function]}
        >
          Þingfundir
        </a>
      </li>
      <li
        className="menu__menu-item"
      >
        <a
          aria-current={null}
          className="menu__menu-link"
          href="/loggjafarthing/1/thingmenn"
          onClick={[Function]}
        >
          Þingmenn
        </a>
      </li>
      <li
        className="menu__menu-item"
      >
        <a
          aria-current={null}
          className="menu__menu-link"
          href="/loggjafarthing/1/thingflokkar"
          onClick={[Function]}
        >
          Þingflokkar
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
